src/Nethermind/Nethermind.Network.Enr/NodeRecordSigner.cs

```diff
@@ -54,53 +54,43 @@ public NodeRecord Deserialize(RlpStream rlpStream)
             {
-                if (key.SequenceEqual(EnrContentKey.IdU8))
-                {
+                case 2 when key.SequenceEqual(EnrContentKey.IdU8):
                     rlpStream.SkipItem();
                     nodeRecord.SetEntry(IdEntry.Instance);
-                }
-                else if (key.SequenceEqual(EnrContentKey.IpU8))
-                {
+                    break;
+                case 2 when key.SequenceEqual(EnrContentKey.IpU8):
                     ReadOnlySpan<byte> ipBytes = rlpStream.DecodeByteArraySpan();
                     IPAddress address = new(ipBytes);
                     nodeRecord.SetEntry(new IpEntry(address));
-                }
-            }
-            else if (key.Length == 3)
-            {
-                if (key.SequenceEqual(EnrContentKey.EthU8))
-                {
+                    break;
+                case 3 when key.SequenceEqual(EnrContentKey.EthU8):
                     _ = rlpStream.ReadSequenceLength();
                     _ = rlpStream.ReadSequenceLength();
                     byte[] forkHash = rlpStream.DecodeByteArray();
                     long nextBlock = rlpStream.DecodeLong();
                     nodeRecord.SetEntry(new EthEntry(forkHash, nextBlock));
-                }
-                else if (key.SequenceEqual(EnrContentKey.TcpU8))
-                {
+                    break;
+                case 3 when key.SequenceEqual(EnrContentKey.TcpU8):
                     int tcpPort = rlpStream.DecodeInt();
                     nodeRecord.SetEntry(new TcpEntry(tcpPort));
-                }
-                else if (key.SequenceEqual(EnrContentKey.UdpU8))
-                {
+                    break;
+                case 3 when key.SequenceEqual(EnrContentKey.UdpU8):
                     int udpPort = rlpStream.DecodeInt();
                     nodeRecord.SetEntry(new UdpEntry(udpPort));
-                }
-            }
-            else if (key.SequenceEqual(EnrContentKey.Secp256K1U8))
-            {
-                ReadOnlySpan<byte> keyBytes = rlpStream.DecodeByteArraySpan();
-                CompressedPublicKey reportedKey = new(keyBytes);
-                nodeRecord.SetEntry(new Secp256K1Entry(reportedKey));
-            }
-            else
-            {
-                // snap
-                canVerify = false;
-                rlpStream.SkipItem();
-                nodeRecord.Snap = true;
+                    break;
+                case 9 when key.SequenceEqual(EnrContentKey.Secp256K1U8):
+                    ReadOnlySpan<byte> keyBytes = rlpStream.DecodeByteArraySpan();
+                    CompressedPublicKey reportedKey = new(keyBytes);
+                    nodeRecord.SetEntry(new Secp256K1Entry(reportedKey));
+                    break;
+                default:
+                    // snap
+                    canVerify = false;
+                    rlpStream.SkipItem();
+                    nodeRecord.Snap = true;
+                    break;
```

Comment:
It is not the same as it was though.

Before if length was 2 or 3 this would never been invoked. Now this will be invoked (but maybe it should?)

Reply:
Was mistake; was invoked in original switch (before the `if` changes)